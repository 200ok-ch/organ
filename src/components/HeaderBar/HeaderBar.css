.header-bar {
    background-color: #5E348C;
    height: 30px;
    display: flex;
    align-items: center;
    padding: 10px;
}

.header-bar__logo {
    width: 45px;
}

.header-bar__title {
    color: white;
    margin: 0;
    margin-left: 5px;
}

.header-bar__back-button {
    color: white;
    text-decoration: none;

    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
}

.header-bar__back-button__directory-path {
    margin-left: 5px;
}

.header-bar__actions {
    margin-left: auto;
    display: flex;
    align-items: center;

    color: white;
}

.header-bar__actions__item {
    color: white;

    margin-left: 20px;
}

.whats-new-icon--has-unseen {
    color: #ff5b5b;
}
