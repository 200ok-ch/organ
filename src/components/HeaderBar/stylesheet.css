@import '../../colors.css';

.header-bar {
  background-color: var(--base2);
  height: 30px;
  padding: 10px;

  display: grid;
  grid-template-columns: 3fr 4fr 3fr;
  align-items: center;

  position: sticky;
  top: 0;
  z-index: 1;
}

.header-bar--with-logo {
  grid-template-columns: 5fr 1fr 5fr;
}

.header-bar__logo-container {
  display: flex;
  align-items: center;
}

.header-bar__logo {
  width: 30px;
}

.header-bar__app-name {
  color: var(--base01);
  margin: 0;
  margin-left: 5px;
}

.header-bar__title {
  color: var(--base01);
  text-align: center;
  font-size: 20px;
  height: 100%;
}

.header-bar__back-button {
  color: var(--base01);
  text-decoration: none;

  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.header-bar__back-button__directory-path {
  margin-left: 5px;
}

.header-bar__actions {
  margin-left: auto;
  display: flex;
  align-items: center;
  cursor: pointer;

  color: var(--base01);
}

.header-bar__actions__item {
  color: var(--base01);

  margin-left: 20px;
}

.settings-icon--has-unseen-changelog {
  color: #ff5b5b;
}

.header-bar__actions__item--disabled {
  opacity: 0.5;
}
