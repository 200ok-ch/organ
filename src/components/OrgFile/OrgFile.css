.org-file__parsing-error-message {
    text-align: center;
    font-size: 16px;

    margin: 10px;
}

.org-file__btn {
    margin: 10px;
    text-align: center;
}

.dirty-indicator {
    padding: 3px;
    background-color: gray;
    opacity: 0.5;
    color: white;
    position: fixed;
    bottom: 100px;
    right: 10px;
    font-weight: bold;
}

.error-message-container {
    padding: 10px;
    font-size: 20px;
}
