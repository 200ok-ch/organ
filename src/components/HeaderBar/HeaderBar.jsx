import React, { Component } from 'react';
import { connect } from 'react-redux';

import logo from './org-web.svg';

import './HeaderBar.css';

class HeaderBar extends Component {
  render() {
    // TODO: sign in button
    // TODO: "whats new" button
    // TODO: settings button

    const { onSignInClick } = this.props;

    return (
      <div className="header-bar">
        <img className="header-bar__logo" src={logo} alt="Logo" width="45" height="45" />
        <h2 className="header-bar__title">org-web</h2>

        <div className="header-bar__actions">
          {/* TODO: show this conditionally */}
          <div className="header-bar__actions__item" onClick={onSignInClick}>Sign in</div>

          <a href="https://github.com/DanielDe/org-web" target="_blank" rel="noopener noreferrer">
            <i className="fab fa-github header-bar__actions__item" />
          </a>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {};
};

const mapDispatchToProps = dispatch => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(HeaderBar);
