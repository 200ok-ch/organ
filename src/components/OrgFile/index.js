import React, { PureComponent, Fragment } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { Redirect } from 'react-router-dom';

import './OrgFile.css';

import HeaderList from './components/HeaderList';
import ActionDrawer from './components/ActionDrawer';

import * as baseActions from '../../actions/base';
import * as dropboxActions from '../../actions/dropbox';

class OrgFile extends PureComponent {
  componentDidMount() {
    const { staticFile, path } = this.props;

    if (!!staticFile) {
      this.props.base.loadStaticFile(staticFile);
    } else if (!!path) {
      this.props.dropbox.downloadFile(path);
    }
  }

  componentWillUnmount() {
    const { staticFile } = this.props;

    if (!!staticFile) {
      this.props.base.unloadStaticFile();
    }
  }

  render() {
    const {
      headers,
      backButtonText,
      onBackClick,
      shouldDisableDirtyIndicator,
      shouldDisableSyncButtons,
      shouldDisableActionDrawer,
      isDirty,
      parsingErrorMessage,
      path,
      staticFile,
    } = this.props;

    if (!path && !staticFile) {
      return <Redirect to="/files" />;
    }

    if (!headers) {
      return <div></div>;
    }

    return (
      <div>
        {headers.size === 0 ? (
          <div className="org-file__parsing-error-message">
            <h3>Couldn't parse file</h3>

            {!!parsingErrorMessage ? (
              <Fragment>{parsingErrorMessage}</Fragment>
            ) : (
              <Fragment>
                If you think this is a bug, please
                {' '}<a href="https://github.com/DanielDe/org-web/issues/new" target="_blank" rel="noopener noreferrer">create an issue</a>
                {' '}and include the org file if possible!
              </Fragment>
            )}
          </div>
        ) : (
          <HeaderList />
        )}

        <div className="btn org-file__btn" onClick={onBackClick}>
          {backButtonText}
        </div>

        {isDirty && !shouldDisableDirtyIndicator && <div className="dirty-indicator">Unpushed changes</div>}

        {!shouldDisableActionDrawer && <ActionDrawer shouldDisableSyncButtons={shouldDisableSyncButtons} />}
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    headers: state.org.present.get('headers'),
    isDirty: state.org.present.get('isDirty'),
  };
};

const mapDispatchToProps = dispatch => {
  return {
    base: bindActionCreators(baseActions, dispatch),
    dropbox: bindActionCreators(dropboxActions, dispatch),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(OrgFile);
