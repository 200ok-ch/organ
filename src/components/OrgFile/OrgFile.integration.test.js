import React from 'react';
import thunk from 'redux-thunk';
import _ from 'lodash';

import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';

import OrgFile from './';
import HeaderBar from '../HeaderBar';

import rootReducer from '../../reducers/';

import { displayFile } from '../../actions/org';

import { Map, fromJS } from 'immutable';
import { formatDistanceToNow } from 'date-fns';

import { render, fireEvent, cleanup, wait } from '@testing-library/react';
afterEach(cleanup);

describe('Render all views', () => {
  jest.mock('react-hotkeys', () => {
    const React = require('react');
    const Fragment = React.Fragment;

    return {
      HotKeys: ({ children }) => <Fragment>{children}</Fragment>,
    };
  });

  const testOrgFile = `
* Top level header
** A nested header
** TODO A todo item with schedule and deadline
   DEADLINE: <2018-10-05 Fri> SCHEDULED: <2019-09-19 Thu>
* Another top level header
Some description content
* A header with tags                                              :tag1:tag2:
* A header with [[https://organice.200ok.ch][a link]]
`;

  let store;

  beforeEach(() => {
    let capture = new Map();
    capture = capture.set('captureTemplates', []);
    store = createStore(
      rootReducer,
      {
        org: {
          past: [],
          present: new Map(),
          future: [],
        },
        syncBackend: Map({
          isAuthenticated: true,
        }),
        capture,
        base: new fromJS({
          customKeybindings: {},
          shouldTapTodoToAdvance: true,
        }),
      },
      applyMiddleware(thunk)
    );
    store.dispatch(displayFile('fixtureTestFile.org', testOrgFile));
  });

  describe('Actions within an Org file', () => {
    test('Can select a header in an org file', () => {
      const { getByText, queryByText } = render(
        <MemoryRouter keyLength={0}>
          <Provider store={store}>
            <OrgFile path="fixtureTestFile.org" />
          </Provider>
        </MemoryRouter>
      );

      expect(queryByText('Scheduled')).toBeFalsy();
      expect(queryByText('Deadline')).toBeFalsy();

      fireEvent.click(getByText('Top level header'));

      expect(queryByText('Scheduled')).toBeTruthy();
      expect(queryByText('Deadline')).toBeTruthy();
    });

    // Org Mode has keywords as workflow states and can cycle through
    // them: https://orgmode.org/manual/Workflow-states.html
    // In organice, we can cycle through them by swiping or by clicking
    // (if enabled). This test checks for the latter.
    test('Can advance todo state for selected header in an org file', () => {
      const { queryByText } = render(
        <MemoryRouter keyLength={0}>
          <Provider store={store}>
            <OrgFile path="fixtureTestFile.org" />
          </Provider>
        </MemoryRouter>
      );

      // In the very beginning, the TODO is hidden, because the file
      // starts folded down to the top level
      expect(queryByText('TODO')).toBeFalsy();

      // Cycle the top header (which will make the next headers
      // [including the TODO] visible
      fireEvent.click(queryByText('Top level header'));

      // In the beginning, the TODO is not DONE
      expect(queryByText('TODO')).toBeTruthy();
      expect(queryByText('DONE')).toBeFalsy();

      // Toggle the TODO
      fireEvent.click(queryByText('TODO'));

      // Then the TODO is DONE
      expect(queryByText('TODO')).toBeFalsy();
      expect(queryByText('DONE')).toBeTruthy();
    });

    test('Can clock in & out of an event', () => {
      const { getByText, queryByText } = render(
        <MemoryRouter keyLength={0}>
          <Provider store={store}>
            <OrgFile path="fixtureTestFile.org" />
          </Provider>
        </MemoryRouter>
      );

      expect(queryByText('Clock In')).toBeFalsy();
      expect(queryByText('Clock Out')).toBeFalsy();

      fireEvent.click(getByText('Top level header'));

      expect(queryByText('Clock In')).toBeTruthy();
      expect(queryByText('Clock Out')).toBeFalsy();
      expect(queryByText(':LOGBOOK:...')).toBeFalsy();

      fireEvent.click(getByText('Clock In'));

      expect(queryByText('Clock In')).toBeFalsy();
      expect(queryByText('Clock Out')).toBeTruthy();
      expect(queryByText(':LOGBOOK:...')).toBeTruthy();

      fireEvent.click(getByText('Clock Out'));

      expect(queryByText('Clock In')).toBeTruthy();
      expect(queryByText('Clock Out')).toBeFalsy();
      expect(queryByText(':LOGBOOK:...')).toBeTruthy();

      fireEvent.click(getByText(':LOGBOOK:...'));

      expect(queryByText('CLOCK:')).toBeTruthy();
      expect(queryByText('=> 0:00')).toBeTruthy();
    });
  });

  describe('Renders everything starting from an Org file', () => {
    let container, getAllByText, getByTitle, getByAltText, getByTestId, queryByText, queryAllByText;
    beforeEach(() => {
      let res = render(
        <MemoryRouter keyLength={0}>
          <Provider store={store}>
            <HeaderBar />
            <OrgFile path="fixtureTestFile.org" />
          </Provider>
        </MemoryRouter>
      );

      container = res.container;
      getAllByText = res.getAllByText;
      getByAltText = res.getByAltText;
      getByTitle = res.getByTitle;
      getByTestId = res.getByTestId;
      queryByText = res.queryByText;
      queryAllByText = res.queryAllByText;
    });

    test('renders an Org file', () => {
      expect(getAllByText(/\*/)).toHaveLength(4);
      expect(container).toMatchSnapshot();
    });

    describe('Undo / Redo', () => {
      test('On loading an Org file, both are disabled', () => {
        expect(getByTitle('Undo').classList.contains('header-bar__actions__item--disabled')).toBe(
          true
        );
        expect(getByTitle('Redo').classList.contains('header-bar__actions__item--disabled')).toBe(
          true
        );
      });

      // FIXME: Why is this test not working?
      test.skip('Undo becomes available on interaction', () => {
        fireEvent.click(queryByText('Top level header'));
        fireEvent.click(queryByText('TODO'));
        // INFO: In the real app, the class is removed now
        expect(getByTitle('Undo').classList.contains('header-bar__actions__item--disabled')).toBe(
          false
        );
      });

      // FIXME: Why is this test not working?
      test.skip('Undo and redo do their respective task', () => {
        fireEvent.click(queryByText('Top level header'));
        expect(queryByText('TODO')).toBeTruthy();
        fireEvent.click(queryByText('TODO'));
        expect(queryByText('TODO')).toBeFalsy();

        // Likely this is what is not working
        fireEvent.click(getByTitle('Undo'));
        // INFO: This is where the test stops working
        expect(queryByText('TODO')).toBeTruthy();

        // fireEvent.click(getByTitle('Redo'));
        // expect(queryByText('TODO')).toBeFalsy();
      });
    });

    describe('Planning items', () => {
      test('deletes planning items', () => {
        fireEvent.click(queryByText('Top level header'));
        fireEvent.click(queryByText('A todo item with schedule and deadline'));
        fireEvent.click(queryByText('<2019-09-19 Thu>'));

        expect(queryByText('SCHEDULED:')).toBeTruthy();
        expect(queryByText('DEADLINE:')).toBeTruthy();

        // First: Delete "Scheduled" time

        // The <input type="date"> field exposes a 'x' button to
        // delete the time. There's no direct API to trigger it.
        // Hence, get the element, remove the time and fire the
        // 'change' event manually.
        let timePicker = getByTestId('timestamp-selector');
        timePicker.value = null;
        fireEvent.change(timePicker);

        expect(queryByText('SCHEDULED:')).toBeFalsy();
        expect(queryByText('DEADLINE:')).toBeTruthy();

        // Second: Delete "Deadline" time

        fireEvent.click(queryByText('<2018-10-05 Fri>'));
        timePicker = getByTestId('timestamp-selector');
        timePicker.value = null;
        fireEvent.change(timePicker);

        expect(queryByText('SCHEDULED:')).toBeFalsy();
        expect(queryByText('DEADLINE:')).toBeFalsy();
      });
    });

    describe('Sharing', () => {
      let windowSpy;
      beforeEach(() => {
        windowSpy = jest.spyOn(global, 'open');
        windowSpy.mockImplementation(x => x);
      });

      afterEach(() => {
        windowSpy.mockRestore();
      });

      test('sends the selected header and its body as an email', () => {
        fireEvent.click(queryByText('Another top level header'));
        fireEvent.click(getByTestId('share'));
        expect(global.open).toBeCalledWith(
          'mailto:?subject=Another%20top%20level%20header&body=%0A%0ASome%20description%20content%0A'
        );
      });
    });

    describe('Agenda', () => {
      test('renders Agenda for an Org file', () => {
        // Agenda is not visible by default
        expect(queryByText('Agenda')).toBeFalsy();
        expect(queryByText('Day')).toBeFalsy();
        expect(queryByText('Month')).toBeFalsy();
        expect(queryByText('A scheduled todo item')).toBeFalsy();

        fireEvent.click(getByTitle('Show agenda'));

        expect(queryByText('Agenda')).toBeTruthy();
        expect(queryByText('Day')).toBeTruthy();
        expect(queryByText('Month')).toBeTruthy();
        expect(queryAllByText('A todo item with schedule and deadline')).toBeTruthy();
      });

      test('Clicking a TODO within the agenda highlights it in the main view', () => {
        fireEvent.click(getByTitle('Show agenda'));
        fireEvent.click(queryAllByText('A todo item with schedule and deadline')[0]);
        expect(queryByText('Agenda')).toBeFalsy();
        expect(queryByText('A todo item with schedule and deadline')).toBeTruthy();
        expect(queryByText('Scheduled')).toBeTruthy();
      });

      test('Clicking the Timestamp in a TODO within the agenda toggles from the date to the time', () => {
        fireEvent.click(getByTitle('Show agenda'));
        const timeSinceScheduled = formatDistanceToNow(new Date('2019-09-19'));
        expect(queryByText(timeSinceScheduled)).toBeFalsy();
        expect(queryByText('09/19')).toBeTruthy();
        fireEvent.click(queryByText('09/19'));
        expect(queryByText('09/19')).toBeFalsy();
        expect(queryByText(`${timeSinceScheduled} ago`)).toBeTruthy();
      });
    });
  });
});
