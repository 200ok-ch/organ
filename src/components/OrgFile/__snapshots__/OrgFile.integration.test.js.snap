// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render all views Org Functionality Renders everything starting from an Org file TaskList orders tasks for an Org file 1`] = `
<div
  class="drawer-inner-container nice-scroll"
  data-testid="drawer"
  style="transform: translateY(768px); height: 92%; overflow: none;"
>
  <div
    class="drawer__grabber"
  />
  <h2
    class="agenda__title"
  >
    Task list
  </h2>
  <datalist
    id="task-list__datalist-filter"
  />
  <div
    class="task-list__input-container"
  >
    <input
      class="textfield task-list__filter-input"
      list="task-list__datalist-filter"
      placeholder="e.g. -DONE doc|man :simple|easy :assignee:nobody|none"
      type="text"
      value=""
    />
  </div>
  <div
    class="task-list__headers-container"
    style="overflow: auto;"
  >
    <div
      class="agenda-day__container"
    >
      <div
        class="agenda-day__headers-container"
      >
        <div
          class="agenda-day__header-container"
        >
          <div
            class="agenda-day__header__header-container"
          >
            <div
              class="title-line"
            >
              <span
                class="todo-keyword"
              >
                TODO
              </span>
              <div
                style="width: 100%;"
              >
                <div
                  style="width: 100%; display: flex; justify-content: space-between;"
                >
                  <span
                    style="word-break: break-word;"
                  >
                    <span>
                      A todo item with schedule and deadline
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="agenda-day__header__planning-item-container"
            >
              <div
                class="task-list__header-planning-type"
              >
                DEADLINE
              </div>
              <div
                class="task-list__header-planning-date task-list__header-planning-date--overdue"
              >
                10/05
              </div>
            </div>
          </div>
        </div>
        <div
          class="agenda-day__header-container"
        >
          <div
            class="agenda-day__header__header-container"
          >
            <div
              class="title-line"
            >
              <span
                class="todo-keyword"
              >
                TODO
              </span>
              <div
                style="width: 100%;"
              >
                <div
                  style="width: 100%; display: flex; justify-content: space-between;"
                >
                  <span
                    style="word-break: break-word;"
                  >
                    <span>
                      A repeating todo
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="agenda-day__header__planning-item-container"
            >
              <div
                class="task-list__header-planning-type"
              >
                SCHEDULED
              </div>
              <div
                class="task-list__header-planning-date task-list__header-planning-date--overdue"
              >
                04/05
              </div>
            </div>
          </div>
        </div>
        <div
          class="agenda-day__header-container"
        >
          <div
            class="agenda-day__header__header-container"
          >
            <div
              class="title-line"
            >
              <span
                class="todo-keyword todo-keyword--done-state"
              >
                FINISHED
              </span>
              <div
                style="width: 100%;"
              >
                <div
                  style="width: 100%; display: flex; justify-content: space-between;"
                >
                  <span
                    style="word-break: break-word;"
                  >
                    <span>
                      A header with a custom todo sequence in DONE state
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div />
          </div>
        </div>
        <div
          class="agenda-day__header-container"
        >
          <div
            class="agenda-day__header__header-container"
          >
            <div
              class="title-line"
            >
              <span
                class="todo-keyword todo-keyword--done-state"
              >
                DONE
              </span>
              <div
                style="width: 100%;"
              >
                <div
                  style="width: 100%; display: flex; justify-content: space-between;"
                >
                  <span
                    style="word-break: break-word;"
                  >
                    <span>
                      A headline that's done a day earlier even
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="agenda-day__header__planning-item-container"
            >
              <div
                class="task-list__header-planning-type"
              >
                SCHEDULED
              </div>
              <div
                class="task-list__header-planning-date task-list__header-planning-date--overdue"
              >
                01/02
              </div>
            </div>
          </div>
        </div>
        <div
          class="agenda-day__header-container"
        >
          <div
            class="agenda-day__header__header-container"
          >
            <div
              class="title-line"
            >
              <span
                class="todo-keyword todo-keyword--done-state"
              >
                DONE
              </span>
              <div
                style="width: 100%;"
              >
                <div
                  style="width: 100%; display: flex; justify-content: space-between;"
                >
                  <span
                    style="word-break: break-word;"
                  >
                    <span>
                      A headline that's done since a loong time
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="agenda-day__header__planning-item-container"
            >
              <div
                class="task-list__header-planning-type"
              >
                SCHEDULED
              </div>
              <div
                class="task-list__header-planning-date task-list__header-planning-date--overdue"
              >
                01/03
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <br />
</div>
`;

exports[`Render all views Org Functionality Renders everything starting from an Org file renders an Org file 1`] = `
<div>
  <div
    class="header-bar"
  >
    <a
      class="header-bar__back-button"
      href="/files/dir1/dir2"
    >
      <i
        class="fas fa-chevron-left"
      />
      <span
        class="header-bar__back-button__directory-path"
      >
        File browser
      </span>
    </a>
    <div
      class="header-bar__title"
    />
    <div
      class="header-bar__actions"
    >
      <i
        class="fas fa-undo header-bar__actions__item header-bar__actions__item--disabled"
        title="Undo"
      />
      <i
        class="fas fa-redo header-bar__actions__item header-bar__actions__item--disabled"
        title="Redo"
      />
      <i
        class="fas fa-question-circle header-bar__actions__item"
        title="Help"
      />
      <div>
        <a
          href="/settings"
        >
          <i
            class="fas fa-cogs header-bar__actions__item"
            title="Settings"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    tabindex="-1"
  >
    <div
      class="org-file-container"
      tabindex="-1"
    >
      <div
        class="header-list-container"
      >
        <div
          class="header"
          style="padding-left: 20px; margin-left: 0px;"
        >
          <div
            class="left-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-check swipe-action-container__icon swipe-action-container__icon--left"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="right-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-times swipe-action-container__icon swipe-action-container__icon--right"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="header__bullet"
            style="margin-left: -16px;"
          >
            *
          </div>
          <div
            class="title-line"
          >
            
            <div
              style="width: 100%;"
            >
              <div
                style="width: 100%; display: flex; justify-content: space-between;"
              >
                <span
                  style="word-break: break-word;"
                >
                  <span>
                    Top level header
                  </span>
                  ...
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="header"
          style="padding-left: 20px; margin-left: 0px;"
        >
          <div
            class="left-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-check swipe-action-container__icon swipe-action-container__icon--left"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="right-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-times swipe-action-container__icon swipe-action-container__icon--right"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="header__bullet"
            style="margin-left: -16px;"
          >
            *
          </div>
          <div
            class="title-line"
          >
            
            <div
              style="width: 100%;"
            >
              <div
                style="width: 100%; display: flex; justify-content: space-between;"
              >
                <span
                  style="word-break: break-word;"
                >
                  <span>
                    Another top level header
                  </span>
                  ...
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="header"
          style="padding-left: 20px; margin-left: 0px;"
        >
          <div
            class="left-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-check swipe-action-container__icon swipe-action-container__icon--left"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="right-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-times swipe-action-container__icon swipe-action-container__icon--right"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="header__bullet"
            style="margin-left: -16px;"
          >
            *
          </div>
          <div
            class="title-line"
          >
            
            <div
              style="width: 100%;"
            >
              <div
                style="width: 100%; display: flex; justify-content: space-between;"
              >
                <span
                  style="word-break: break-word;"
                >
                  <span>
                    A header with tags                                              
                  </span>
                  
                </span>
              </div>
              <div>
                <div
                  class="header-tag"
                >
                  tag1
                </div>
                <div
                  class="header-tag"
                >
                  tag2
                </div>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="header"
          style="padding-left: 20px; margin-left: 0px;"
        >
          <div
            class="left-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-check swipe-action-container__icon swipe-action-container__icon--left"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="right-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-times swipe-action-container__icon swipe-action-container__icon--right"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="header__bullet"
            style="margin-left: -16px;"
          >
            *
          </div>
          <div
            class="title-line"
          >
            
            <div
              style="width: 100%;"
            >
              <div
                style="width: 100%; display: flex; justify-content: space-between;"
              >
                <span
                  style="word-break: break-word;"
                >
                  <span>
                    A header with 
                    <a
                      href="https://organice.200ok.ch"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      a link
                    </a>
                  </span>
                  
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="header"
          style="padding-left: 20px; margin-left: 0px;"
        >
          <div
            class="left-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-check swipe-action-container__icon swipe-action-container__icon--left"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="right-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-times swipe-action-container__icon swipe-action-container__icon--right"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="header__bullet"
            style="margin-left: -16px;"
          >
            *
          </div>
          <div
            class="title-line"
          >
            
            <div
              style="width: 100%;"
            >
              <div
                style="width: 100%; display: flex; justify-content: space-between;"
              >
                <span
                  style="word-break: break-word;"
                >
                  <span>
                    A header with various links as content
                  </span>
                  ...
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="header"
          style="padding-left: 20px; margin-left: 0px;"
        >
          <div
            class="left-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-check swipe-action-container__icon swipe-action-container__icon--left"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="right-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-times swipe-action-container__icon swipe-action-container__icon--right"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="header__bullet"
            style="margin-left: -16px;"
          >
            *
          </div>
          <div
            class="title-line"
          >
            
            <div
              style="width: 100%;"
            >
              <div
                style="width: 100%; display: flex; justify-content: space-between;"
              >
                <span
                  style="word-break: break-word;"
                >
                  <span>
                    A header with a URL, mail address and phone number as content
                  </span>
                  ...
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="header"
          style="padding-left: 20px; margin-left: 0px;"
        >
          <div
            class="left-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-check swipe-action-container__icon swipe-action-container__icon--left"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="right-swipe-action-container"
            style="width: 0px; background-color: rgb(211, 211, 211);"
          >
            <i
              class="fas fa-times swipe-action-container__icon swipe-action-container__icon--right"
              style="display: none; color: rgb(0, 0, 0);"
            />
          </div>
          <div
            class="header__bullet"
            style="margin-left: -16px;"
          >
            *
          </div>
          <div
            class="title-line"
          >
            <span
              class="todo-keyword todo-keyword--done-state"
            >
              FINISHED
            </span>
            <div
              style="width: 100%;"
            >
              <div
                style="width: 100%; display: flex; justify-content: space-between;"
              >
                <span
                  style="word-break: break-word;"
                >
                  <span>
                    A header with a custom todo sequence in DONE state
                  </span>
                  
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
      </div>
      <div
        class="action-drawer-container nice-scroll"
      >
        <button
          class="btn btn--circle action-drawer__btn fas fa-lg fa-cloud action-drawer__btn--with-sub-icon"
          style="opacity: 1;"
          title="Sync changes"
        >
          <i
            class="fas fa-xs fa-sync-alt action-drawer__btn__sub-icon"
          />
        </button>
        <button
          class="btn btn--circle action-drawer__btn fas fa-lg fa-calendar-alt"
          style="opacity: 1;"
          title="Show agenda"
        />
        <div
          class="action-drawer__arrow-buttons-container"
          style="left: 0px;"
        >
          <button
            class="btn btn--circle action-drawer__btn action-drawer__arrow-button fas fa-lg fa-arrow-up"
            style="opacity: 1; box-shadow: none; bottom: 0px;"
            title="Move header up"
          />
          <button
            class="btn btn--circle action-drawer__btn action-drawer__arrow-button fas fa-lg fa-arrow-down"
            style="opacity: 1; box-shadow: none; bottom: 0px;"
            title="Move header down"
          />
          <button
            class="btn btn--circle action-drawer__btn action-drawer__arrow-button fas fa-lg fa-arrow-left"
            style="opacity: 1; box-shadow: none; bottom: 0px; right: 0px;"
            title="Move header left"
          />
          <button
            class="btn btn--circle action-drawer__btn action-drawer__arrow-button fas fa-lg fa-arrow-right"
            style="opacity: 1; box-shadow: none; bottom: 0px; left: 0px;"
            title="Move header right"
          />
          <button
            class="btn btn--circle action-drawer__btn action-drawer__arrow-button fas fa-lg fa-chevron-left"
            style="opacity: 1; box-shadow: none; bottom: 0px; right: 0px;"
            title="Move entire subtree left"
          />
          <button
            class="btn btn--circle action-drawer__btn action-drawer__arrow-button fas fa-lg fa-chevron-right"
            style="opacity: 1; box-shadow: none; bottom: 0px; left: 0px;"
            title="Move entire subtree right"
          />
          <button
            class="btn btn--circle action-drawer__btn action-drawer__main-arrow-button fas fa-lg fa-arrows-alt"
            style="opacity: 1;"
            title="Show movement buttons"
          />
        </div>
        <div
          class="action-drawer__capture-buttons-container"
        >
          <button
            class="btn btn--circle action-drawer__btn fas fa-lg fa-search"
            style="opacity: 1; position: relative; z-index: 1;"
            title="Show Search / Task List"
          />
          <button
            class="btn btn--circle action-drawer__btn fas fa-lg fa-search"
            style="position: absolute; z-index: 0; left: 0px; opacity: 1; box-shadow: none; bottom: 0px;"
            title="Show search"
          />
          <button
            class="btn btn--circle action-drawer__btn fas fa-lg fa-tasks"
            style="position: absolute; z-index: 0; left: 0px; opacity: 1; box-shadow: none; bottom: 0px;"
            title="Show task list"
          />
        </div>
        <div
          class="action-drawer__capture-buttons-container"
        >
          <button
            class="btn btn--circle action-drawer__btn fas fa-lg fa-plus"
            style="opacity: 1; position: relative; z-index: 1;"
            title="Show capture templates"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
